core/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -161,6 +169,10 @@ public T omitNorms(boolean omitNorms) {
         }
 
         public T indexOptions(IndexOptions indexOptions) {
+            if (allowsIndexOptions() == false) {
+                throw new UnsupportedOperationException(
+                        "index_options not allowed for field type [" + fieldType().typeName() + "]");
+            }
```

Comment:
Could we override the `indexOptions` method in the number field type instead of introducing a new method?

Reply:
Sure, I changed this. Will wait for another CI run before I merge.